pilot/pkg/config/kube/gateway/backend_policies.go

```diff
@@ -0,0 +1,423 @@
+			pols := slices.SortFunc(i.Objects, func(a, b BackendPolicy) int {
+				if r := a.CreationTime.Compare(b.CreationTime); r != 0 {
+					return r
+				}
+				if r := cmp.Compare(a.Source.Namespace, b.Source.Namespace); r != 0 {
+					return r
+				}
+				return cmp.Compare(a.Source.Name, b.Source.Name)
+			})
+			_ = pols
+			tlsSet := false
+			lbSet := false
+			spec := &networking.DestinationRule{
+				Host:          fmt.Sprintf("%s.%s.svc.%v", svc.Name, svc.Namespace, domainSuffix),
+				TrafficPolicy: &networking.TrafficPolicy{},
+			}
+			parents := make([]string, 0, len(pols))
+			for _, pol := range pols {
+				parents = append(parents, fmt.Sprintf("%s/%s.%s", pol.Source.Kind, pol.Source.Namespace, pol.Source.Name))
+				if pol.TLS != nil {
+					if tlsSet {
+						// We only allow 1. TODO: report status if there are multiple
+						continue
+					}
+					tlsSet = true
+					spec.TrafficPolicy.Tls = pol.TLS
+				}
+				if pol.LoadBalancer != nil {
+					if lbSet {
+						// We only allow 1. TODO: report status if there are multiple
+						continue
+					}
+					lbSet = true
+					spec.TrafficPolicy.LoadBalancer = pol.LoadBalancer
+				}
+			}
+			cfg := &config.Config{
+				Meta: config.Meta{
+					GroupVersionKind: gvk.DestinationRule,
+					Name:             fmt.Sprintf("%s-%s", svc.Name, constants.KubernetesGatewayName),
+					Namespace:        svc.Namespace,
+					Annotations: map[string]string{
+						constants.InternalParentNames: strings.Join(parents, ","),
+					},
+				},
+				Spec: spec,
+			}
+			return &cfg
+		}, opts.WithName("BackendPolicyMerged")...)
+	return merged
+}
+
+func BackendTLSPolicyCollection(
+	tlsPolicies krt.Collection[*gatewayalpha3.BackendTLSPolicy],
+	references *ReferenceSet,
+	opts krt.OptionsBuilder,
+) (krt.StatusCollection[*gatewayalpha3.BackendTLSPolicy, gatewayalpha2.PolicyStatus], krt.Collection[BackendPolicy]) {
+	return krt.NewStatusManyCollection(tlsPolicies, func(ctx krt.HandlerContext, i *gatewayalpha3.BackendTLSPolicy) (
+		*gatewayalpha2.PolicyStatus,
+		[]BackendPolicy,
+	) {
+		status := i.Status.DeepCopy()
+		res := make([]BackendPolicy, 0, len(i.Spec.TargetRefs))
+		ancestors := make([]gatewayalpha2.PolicyAncestorStatus, 0, len(i.Spec.TargetRefs))
+
+		tls := &networking.ClientTLSSettings{Mode: networking.ClientTLSSettings_SIMPLE}
+		s := i.Spec
+
+		conds := map[string]*condition{
+			string(gatewayalpha2.PolicyConditionAccepted): {
+				reason:  string(gatewayalpha2.PolicyReasonAccepted),
+				message: "Configuration is valid",
+			},
+		}
+		tls.Sni = string(s.Validation.Hostname)
+		tls.SubjectAltNames = slices.MapFilter(s.Validation.SubjectAltNames, func(e gatewayalpha3.SubjectAltName) *string {
+			switch e.Type {
+			case gatewayalpha3.HostnameSubjectAltNameType:
+				return ptr.Of(string(e.Hostname))
+			case gatewayalpha3.URISubjectAltNameType:
+				return ptr.Of(string(e.URI))
+			}
+			return nil
+		})
+		if wk := s.Validation.WellKnownCACertificates; wk != nil {
```

Comment:
nit: maybe extract to a helper function so you can use returns and breaks to make the code cleaner?